File: app/src/main/java/ru/buylist/data/dao/CategoryDao.java
package ru.buylist.data.dao;


import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import ru.buylist.data.entity.Category;

@Dao
public interface CategoryDao {

    @Insert
    void addCategory(Category category);

    @Delete
    void deleteCategory(Category category);

    @Update
    void updateCategory(Category category);

    @Query("SELECT * FROM categories WHERE name = :name")
    Category getCategory(String name);

    @Query("SELECT * FROM categories")
    List<Category> getAllCategories();

    @Query("SELECT * FROM categories")
    LiveData<List<Category>> getLiveCategories();
}

File: app/src/main/java/ru/buylist/data/dao/CategoryDao.kt
package ru.buylist.data.dao

import androidx.room.*
import ru.buylist.data.entity.Category

@Dao
interface CategoryDao {

    @Insert
    fun insertCategory(category: Category)

    @Update
    fun updateCategory(category: Category)

    @Delete
    fun deleteCategory(category: Category)

    @Delete
    fun deleteSelectedCategory(categories: MutableList<Category>)

    @Query("DELETE FROM categories")
    fun deleteAllCategories()

    @Query("SELECT * FROM categories")
    fun getCategories(): MutableList<Category>

    @Query("SELECT * FROM categories WHERE name = :categoryName")
    fun getCategory(categoryName: String)

    @Query("UPDATE categories SET color = :categoryColor")
    fun updateCategoryColor(categoryColor: String)

}